src/osd/SnapMapper.cc

```diff
@@ -764,6 +764,10 @@ int SnapMapper::_lookup_purged_snap(
   decode(gotpool, p);
   decode(*begin, p);
   decode(*end, p);
+  if (gotpool != pool) {
```

Comment:
not sure I understand how it happens that we query with a key that contains the correct pool, and still get something else. This should probably be clarified in the commit message.

Reply:
We aren't "querying a key" -- we're invoking get_next_or_current.  If the specified key doesn't exist and there are no keys which sort after that point in the same pool, we'll get the first key of the next pool.  If that key happens to be >= begin and < *end, it'll pass check immediately after.